openedx/features/content_type_gating/field_override.py

```diff
@@ -44,11 +44,21 @@ def get(self, block, name, default):
             if group_access_field is not None:
                 original_group_access = group_access_field.from_json(raw_value)
 
+        group_ids = [settings.CONTENT_TYPE_GATE_GROUP_IDS['full_access']]
         if original_group_access is None:
             original_group_access = {}
+
+            # For Feature Based Enrollments, we want to allow granting access
+            # to all graded problems in a unit at the unit level
+            parent = block.get_parent()
+            if parent.scope_ids.block_type == 'vertical':
```

Comment:
This shouldn't hard-code `vertical` (see above).

Also, can we maybe just check if `CONTENT_TYPE_GATE_GROUP_IDS` is in `block.merged_group_access` rather than in `block.fields.get('group_access')`? I'd rather not replicate the `merged_group_access` logic if we don't have to.